account_invoice_import_invoice2data/wizard/account_invoice_import.py

```diff
@@ -90,9 +90,79 @@ def invoice2data_to_parsed_inv(self, invoice2data_res):
             'date_start': invoice2data_res.get('date_start'),
             'date_end': invoice2data_res.get('date_end'),
             'description': invoice2data_res.get('description'),
+            'mapping_lines': invoice2data_res.get('lines', [])
             }
         if 'amount_untaxed' in invoice2data_res:
             parsed_inv['amount_untaxed'] = invoice2data_res['amount_untaxed']
         if 'amount_tax' in invoice2data_res:
             parsed_inv['amount_tax'] = invoice2data_res['amount_tax']
         return parsed_inv
+
+    @api.model
+    def _prepare_create_invoice_vals(self, parsed_inv):
+        """ Extend invoice with line mapping. """
+        vals = super(AccountInvoiceImport, self)._prepare_create_invoice_vals(
+            parsed_inv)
+        partner = self.env['res.partner'].browse(vals['partner_id'])
+        template = self.env['invoice2data.template'].search([
+            ('related_partner', '=', partner.id)
+        ])
+        config = partner.invoice_import_id
+        if (config.invoice_line_method == 'product_mapping' and
+                parsed_inv.get('mapping_lines')):
```

Comment:
I don't understand the data model choice here. There is a partner config that determines the default ledger and analytic accounts for the partner. Then there are templates that determine the result of the parsing process. Those templates are not customer specific. At this point, it is not clear what template was used to extract the data (and presumably it is not important).

You have chosen to link the product mappings to the template, but the templates you use here do not depend on the template. They depend on the partner. Therefore, I think you might want to link the product mappings to the import config (that you already retrieve by partner here) rather than to the template, especially because the product mappings help determine the ledger and analytic accounts, overriding the settings from the template so they are related in that way as well.